<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from rob1_gripper.urdf.xacro        | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="rob1">
  <link name="rob1_real_base_link"/>
  <joint name="rob1_real_base_joint" type="fixed">
    <parent link="rob1_real_base_link"/>
    <child link="rob1_base_link"/>
    <origin rpy="0.0 0.0 3.14159" xyz="0 0.0 0.0"/>
  </joint>
  <link name="rob1_world"/>
  <joint name="rob1_world_base_joint" type="fixed">
    <parent link="rob1_world"/>
    <child link="rob1_real_base_link"/>
    <origin rpy="0.0 0.0 0" xyz="0 0.0 0.0"/>
  </joint>
  <joint name="rob1_ur_real_ee_fixed_joint" type="fixed">
    <parent link="rob1_ee_link"/>
    <child link="rob1_real_ee_link"/>
    <origin rpy="-1.570795 0.0 -1.570795" xyz="0.0 0.0 0.0"/>
  </joint>
  <link name="rob1_real_ee_link"/>
  <!--property name="shoulder_height" value="0.089159" /-->
  <!--property name="shoulder_offset" value="0.13585" /-->
  <!-- shoulder_offset - elbow_offset + wrist_1_length = 0.10915 -->
  <!--property name="upper_arm_length" value="0.42500" /-->
  <!--property name="elbow_offset" value="0.1197" /-->
  <!-- CAD measured -->
  <!--property name="forearm_length" value="0.39225" /-->
  <!--property name="wrist_1_length" value="0.093" /-->
  <!-- CAD measured -->
  <!--property name="wrist_2_length" value="0.09465" /-->
  <!-- In CAD this distance is 0.930, but in the spec it is 0.09465 -->
  <!--property name="wrist_3_length" value="0.0823" /-->
  <!-- manually measured -->
  <link name="rob1_base_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/base.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/base.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="4.0"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.00443333156" ixy="0.0" ixz="0.0" iyy="0.00443333156" iyz="0.0" izz="0.0072"/>
    </inertial>
  </link>
  <joint name="rob1_shoulder_pan_joint" type="revolute">
    <parent link="rob1_base_link"/>
    <child link="rob1_shoulder_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.163"/>
    <axis xyz="0 0 1"/>
    <limit effort="150.0" lower="-6.28318530718" upper="6.28318530718" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_shoulder_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/shoulder.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/shoulder.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="3.7"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <inertia ixx="0.010267495893" ixy="0.0" ixz="0.0" iyy="0.010267495893" iyz="0.0" izz="0.00666"/>
    </inertial>
  </link>
  <joint name="rob1_shoulder_lift_joint" type="revolute">
    <parent link="rob1_shoulder_link"/>
    <child link="rob1_upper_arm_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.138 0.0"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-6.28318530718" upper="6.28318530718" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_upper_arm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/upperarm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/upperarm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="8.393"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.2125"/>
      <inertia ixx="0.133885781862" ixy="0.0" ixz="0.0" iyy="0.133885781862" iyz="0.0" izz="0.0151074"/>
    </inertial>
  </link>
  <joint name="rob1_elbow_joint" type="revolute">
    <parent link="rob1_upper_arm_link"/>
    <child link="rob1_forearm_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 -0.131 0.425"/>
    <axis xyz="0 1 0"/>
    <limit effort="150.0" lower="-3.14159265359" upper="3.14159265359" velocity="3.14"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_forearm_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/forearm.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/forearm.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="2.275"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.196"/>
      <inertia ixx="0.0311796208615" ixy="0.0" ixz="0.0" iyy="0.0311796208615" iyz="0.0" izz="0.004095"/>
    </inertial>
  </link>
  <joint name="rob1_wrist_1_joint" type="revolute">
    <parent link="rob1_forearm_link"/>
    <child link="rob1_wrist_1_link"/>
    <origin rpy="0.0 1.57079632679 0.0" xyz="0.0 0.0 0.392"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_wrist_1_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist1.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist1.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.127 0.0"/>
      <inertia ixx="0.00255989897604" ixy="0.0" ixz="0.0" iyy="0.00255989897604" iyz="0.0" izz="0.0021942"/>
    </inertial>
  </link>
  <joint name="rob1_wrist_2_joint" type="revolute">
    <parent link="rob1_wrist_1_link"/>
    <child link="rob1_wrist_2_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.127 0.0"/>
    <axis xyz="0 0 1"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_wrist_2_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist2.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist2.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.219"/>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.1"/>
      <inertia ixx="0.00255989897604" ixy="0.0" ixz="0.0" iyy="0.00255989897604" iyz="0.0" izz="0.0021942"/>
    </inertial>
  </link>
  <joint name="rob1_wrist_3_joint" type="revolute">
    <parent link="rob1_wrist_2_link"/>
    <child link="rob1_wrist_3_link"/>
    <origin rpy="0.0 0.0 0.0" xyz="0.0 0.0 0.1"/>
    <axis xyz="0 1 0"/>
    <limit effort="28.0" lower="-6.28318530718" upper="6.28318530718" velocity="6.28"/>
    <dynamics damping="0.0" friction="0.0"/>
  </joint>
  <link name="rob1_wrist_3_link">
    <visual>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/visual/wrist3.dae"/>
      </geometry>
      <material name="LightGrey">
        <color rgba="0.7 0.7 0.7 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://ur_e_description/meshes/ur5e/collision/wrist3.stl"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1879"/>
      <origin rpy="1.57079632679 0 0" xyz="0.0 0.0771 0.0"/>
      <inertia ixx="9.89041005217e-05" ixy="0.0" ixz="0.0" iyy="9.89041005217e-05" iyz="0.0" izz="0.0001321171875"/>
    </inertial>
  </link>
  <joint name="rob1_ee_fixed_joint" type="fixed">
    <parent link="rob1_wrist_3_link"/>
    <child link="rob1_ee_link"/>
    <origin rpy="0.0 0.0 1.57079632679" xyz="0.0 0.1 0.0"/>
  </joint>
  <link name="rob1_ee_link">
    <collision>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
      <origin rpy="0 0 0" xyz="-0.01 0 0"/>
    </collision>
  </link>
  <transmission name="rob1_shoulder_pan_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_shoulder_pan_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="rob1_shoulder_lift_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_shoulder_lift_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_shoulder_lift_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="rob1_elbow_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_elbow_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="rob1_wrist_1_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_wrist_1_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_wrist_1_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="rob1_wrist_2_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_wrist_2_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_wrist_2_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="rob1_wrist_3_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_wrist_3_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_wrist_3_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="rob1_shoulder_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_upper_arm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_forearm_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_wrist_1_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_wrist_3_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_wrist_2_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <gazebo reference="rob1_ee_link">
    <selfCollide>true</selfCollide>
  </gazebo>
  <!-- ROS base_link to UR 'Base' Coordinates transform -->
  <link name="rob1_base"/>
  <joint name="rob1_base_link-base_fixed_joint" type="fixed">
    <!-- NOTE: this rotation is only needed as long as base_link itself is
                 not corrected wrt the real robot (ie: rotated over 180
                 degrees)
      -->
    <origin rpy="0 0 -3.14159265359" xyz="0 0 0"/>
    <parent link="rob1_base_link"/>
    <child link="rob1_base"/>
  </joint>
  <!-- Frame coincident with all-zeros TCP on UR controller -->
  <link name="rob1_tool0"/>
  <joint name="rob1_wrist_3_link-tool0_fixed_joint" type="fixed">
    <origin rpy="-1.57079632679 0 0" xyz="0 0.1 0"/>
    <parent link="rob1_wrist_3_link"/>
    <child link="rob1_tool0"/>
  </joint>
  <joint name="rob1_ur_gripper_fixed_joint" type="fixed">
    <parent link="rob1_tool0"/>
    <child link="rob1_robotiq_arg2f_base_link"/>
    <origin rpy="0 0 -1.570795" xyz="0 0 0"/>
  </joint>
  <link name="rob1_robotiq_arg2f_base_link">
    <inertial>
      <origin rpy="0 0 0" xyz="8.625E-08 -4.6583E-06 0.03145"/>
      <mass value="0.22652"/>
      <inertia ixx="0.00020005" ixy="-4.2442E-10" ixz="-2.9069E-10" iyy="0.00017832" iyz="-3.4402E-08" izz="0.00013478"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_base_link.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_base_link.stl"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_left_outer_knuckle">
    <!--<inertial>
        <origin xyz="-0.000200000000003065 0.0199435877845359 0.0292245259211331" rpy="0 0 0" />
        <mass value="0.00853198276973456" />
        <inertia ixx="2.89328108496468E-06" ixy="-1.57935047237397E-19" ixz="-1.93980378593255E-19" iyy="1.86719750325683E-06" iyz="-1.21858577871576E-06" izz="1.21905238907251E-06" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_outer_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_outer_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_left_outer_finger">
    <!--<inertial>
        <origin xyz="0.00030115855001899 0.0373907951953854 -0.0208027427000385" rpy="0 0 0" />
        <mass value="0.022614240507152" />
        <inertia ixx="1.52518312458174E-05" ixy="9.76583423954399E-10" ixz="-5.43838577022588E-10" iyy="6.17694243867776E-06" iyz="6.78636130740228E-06" izz="1.16494917907219E-05" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_outer_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_outer_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_left_inner_finger">
    <!--<inertial>
        <origin xyz="0.000299999999999317 0.0160078233491243 -0.0136945669206257" rpy="0 0 0" />
        <mass value="0.0104003125914103" />
        <inertia ixx="2.71909453810972E-06" ixy="1.35402465472579E-21" ixz="-7.1817349065269E-22" iyy="7.69100314106116E-07" iyz="6.74715432769696E-07" izz="2.30315190420171E-06" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_inner_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_inner_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_left_inner_finger_pad">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <box size="0.022 0.00635 0.0375"/>
      </geometry>
      <material name="">
        <color rgba="0.9 0.9 0.9 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <box size="0.022 0.00635 0.0375"/>
      </geometry>
      <material name="">
        <color rgba="0.9 0.0 0.0 1"/>
      </material>
    </collision>
  </link>
  <link name="rob1_left_inner_knuckle">
    <!--<inertial>
        <origin xyz="0.000123011831763771 0.0507850843201817 0.00103968640075166" rpy="0 0 0" />
       <mass value="0.0271177346495152" />
        <inertia ixx="2.61910379223783E-05" ixy="-2.43616858946494E-07" ixz="-6.37789906117123E-09" iyy="2.8270243746167E-06" iyz="-5.37200748039765E-07" izz="2.83695868220296E-05" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_inner_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_inner_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_right_outer_knuckle">
    <!--<inertial>
        <origin xyz="-0.000200000000003065 0.0199435877845359 0.0292245259211331" rpy="0 0 0" />
        <mass value="0.00853198276973456" />
        <inertia ixx="2.89328108496468E-06" ixy="-1.57935047237397E-19" ixz="-1.93980378593255E-19" iyy="1.86719750325683E-06" iyz="-1.21858577871576E-06" izz="1.21905238907251E-06" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_outer_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_outer_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_right_outer_finger">
    <!--<inertial>
        <origin xyz="0.00030115855001899 0.0373907951953854 -0.0208027427000385" rpy="0 0 0" />
        <mass value="0.022614240507152" />
        <inertia ixx="1.52518312458174E-05" ixy="9.76583423954399E-10" ixz="-5.43838577022588E-10" iyy="6.17694243867776E-06" iyz="6.78636130740228E-06" izz="1.16494917907219E-05" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_outer_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_outer_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_right_inner_finger">
    <!--<inertial>
        <origin xyz="0.000299999999999317 0.0160078233491243 -0.0136945669206257" rpy="0 0 0" />
        <mass value="0.0104003125914103" />
        <inertia ixx="2.71909453810972E-06" ixy="1.35402465472579E-21" ixz="-7.1817349065269E-22" iyy="7.69100314106116E-07" iyz="6.74715432769696E-07" izz="2.30315190420171E-06" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_inner_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_inner_finger.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <link name="rob1_right_inner_finger_pad">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <box size="0.022 0.00635 0.0375"/>
      </geometry>
      <material name="">
        <color rgba="0.9 0.9 0.9 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <box size="0.022 0.00635 0.0375"/>
      </geometry>
      <material name="">
        <color rgba="0.9 0.0 0.0 1"/>
      </material>
    </collision>
  </link>
  <link name="rob1_right_inner_knuckle">
    <!--<inertial>
        <origin xyz="0.000123011831763771 0.0507850843201817 0.00103968640075166" rpy="0 0 0" />
       <mass value="0.0271177346495152" />
        <inertia ixx="2.61910379223783E-05" ixy="-2.43616858946494E-07" ixz="-6.37789906117123E-09" iyy="2.8270243746167E-06" iyz="-5.37200748039765E-07" izz="2.83695868220296E-05" />
        </inertial> -->
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/visual/robotiq_arg2f_85_inner_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
      <material name="">
        <color rgba="0.1 0.1 0.1 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://robotiq_2f_85_gripper_visualization/meshes/collision/robotiq_arg2f_85_inner_knuckle.dae" scale="0.001 0.001 0.001"/>
      </geometry>
    </collision>
  </link>
  <joint name="rob1_finger_joint" type="revolute">
    <origin rpy="0 0 3.14159265359" xyz="0 -0.0306011 0.054904"/>
    <parent link="rob1_robotiq_arg2f_base_link"/>
    <child link="rob1_left_outer_knuckle"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.8" velocity="2.0"/>
  </joint>
  <joint name="rob1_left_outer_finger_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0315 -0.0041"/>
    <parent link="rob1_left_outer_knuckle"/>
    <child link="rob1_left_outer_finger"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="rob1_left_inner_knuckle_joint" type="revolute">
    <!-- <origin xyz="0 ${reflect * -0.0127} 0.06142" rpy="${pi / 2 + .725} 0 ${(reflect - 1) * pi / 2}" /> -->
    <origin rpy="0 0 3.14159265359" xyz="0 -0.0127 0.06142"/>
    <parent link="rob1_robotiq_arg2f_base_link"/>
    <child link="rob1_left_inner_knuckle"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.8757" velocity="2.0"/>
    <mimic joint="rob1_finger_joint" multiplier="1" offset="0"/>
  </joint>
  <joint name="rob1_left_inner_finger_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0061 0.0471"/>
    <parent link="rob1_left_outer_finger"/>
    <child link="rob1_left_inner_finger"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.8757" velocity="2.0"/>
    <mimic joint="rob1_finger_joint" multiplier="-1" offset="0"/>
  </joint>
  <joint name="rob1_left_inner_finger_pad_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.0220203446692936 0.03242"/>
    <parent link="rob1_left_inner_finger"/>
    <child link="rob1_left_inner_finger_pad"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="rob1_right_outer_knuckle_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0306011 0.054904"/>
    <parent link="rob1_robotiq_arg2f_base_link"/>
    <child link="rob1_right_outer_knuckle"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.81" velocity="2.0"/>
    <mimic joint="rob1_finger_joint" multiplier="1" offset="0"/>
  </joint>
  <joint name="rob1_right_outer_finger_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0.0315 -0.0041"/>
    <parent link="rob1_right_outer_knuckle"/>
    <child link="rob1_right_outer_finger"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="rob1_right_inner_knuckle_joint" type="revolute">
    <!-- <origin xyz="0 ${reflect * -0.0127} 0.06142" rpy="${pi / 2 + .725} 0 ${(reflect - 1) * pi / 2}" /> -->
    <origin rpy="0 0 0.0" xyz="0 0.0127 0.06142"/>
    <parent link="rob1_robotiq_arg2f_base_link"/>
    <child link="rob1_right_inner_knuckle"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.8757" velocity="2.0"/>
    <mimic joint="rob1_finger_joint" multiplier="1" offset="0"/>
  </joint>
  <joint name="rob1_right_inner_finger_joint" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.0061 0.0471"/>
    <parent link="rob1_right_outer_finger"/>
    <child link="rob1_right_inner_finger"/>
    <axis xyz="1 0 0"/>
    <limit effort="1000" lower="0" upper="0.8757" velocity="2.0"/>
    <mimic joint="rob1_finger_joint" multiplier="-1" offset="0"/>
  </joint>
  <joint name="rob1_right_inner_finger_pad_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 -0.0220203446692936 0.03242"/>
    <parent link="rob1_right_inner_finger"/>
    <child link="rob1_right_inner_finger_pad"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="rob1_finger_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="rob1_finger_joint">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="rob1_finger_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
</robot>

